Makefile

```diff
@@ -39,8 +39,8 @@ cleanup-kind:
 	kind delete cluster --name=flux-e2e-test
 	rm $(TEST_KUBECONFIG)
 
-test: $(EMBEDDED_MANIFESTS_TARGET) tidy fmt vet
-	go test ./... -coverprofile cover.out --tags=unit
+test: $(EMBEDDED_MANIFESTS_TARGET) tidy fmt vet setup-envtest
+	KUBEBUILDER_ASSETS=$(shell $(SETUP_ENVTEST) use -p path) go test ./... -coverprofile cover.out --tags=unit
```

Comment:
Please see https://github.com/fluxcd/pkg/blob/main/Makefile#L6 as it is now it will fail on macOS.

Reply:
I'm going to need more hand holding here.

I copied this from that file, but am not sure how SETUP_ENVTEST is set there and how to reconcile this with the setup envtest lines above

Reply:
IF you undo the changes to Makefile and rebase with main it should be good.

Reply:
Thanks! I'll fix the e2e error also and rebase everything.

Can you explain how SETUP_ENVTEST is set outside of the GitHub action when running locally? Is `setup_envtest:` running implicitly somewhere to set that variable at the start?

Reply:
On your local machine you would just run `make test` and it should install the envtest binaries then run the unit tests. For e2e tests, you run first `make setup-kind` then `make e2e`, this will leave the kind cluster running so you debug things or run `make test-with-kind` that will create a cluster each time, run e2e then delete the cluster.

Reply:
Right, that was the case with this PR too. To clarify, I'm not asking a workflow question, I'm asking a makefile technical question on how the variable is set to increase my understanding (I'm trying to not blindly just copy the code from the other package.). Thanks for your patience.

Reply:
So `make test` runs `setup-envtest` then runs the `go test`.

Reply:
I guess my confusion is that `KUBEBUILDER_ASSETS` is using a variable at the top of the file before `setup-env` even runs. I'm guessing it must not evaluated until its used, and that is what I was missing...